verifier/verifier.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * ZCrypto Copyright 2017 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package verifier
+
+import (
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+)
+
+// VerificationResult contains the result of a verification of a certificate
+// against a browser. It discerns between NameError, meaning the name on the
+// certificate does not match, and a ValidationError, meaning some sort of issue
+// with the cryptography, or other issues with the chain.
+type VerificationResult struct {
+
+	// The name being checked, e.g. "www.censys.io".
+	Name string
+
+	// True if the certificate is whitelisted by a given browser as trusted, e.g.
+	// when Apple whitelisted subset of StartCom certs by SPKI hash in the wake of
+	// the WoSign incidents surrounding misissuance.
+	Whitelisted bool
```

Comment:
Why are we returning this? It doesn't make sense as a return result of this level call? What happens for example if something is both whitelisted and blacklisted?